### PR TITLE
Speed up autoupdates

### PR DESCRIPTION
**Pull request by emanuelschuetze**
Use 'bulk mode' for DS.inject by calling inject
only once per each collection.

### CODE REVIEW

File: openslides/core/static/js/core/base.js [hunk 1]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
             } catch(err) {
                 console.error(json);
             }
-            _.forEach(dataList, function(data) {
-                console.log("Received object: " + data.collection + ", " + data.id);
-                var instance = DS.get(data.collection, data.id);
-                if (data.action == 'changed') {
+
+            var dataList_byCollection = _.groupBy(dataList, 'collection');
+            var counter = 0;
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: This line is not used.

File: openslides/core/static/js/core/base.js [hunk 2]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
...
-                        // The instance is in the local db
                         dsEject(data.collection, instance);
                     }
-                    DS.inject(data.collection, data.data);
-                } else if (data.action == 'deleted') {
-                    if (instance) {
-                        // The instance is in the local db
-                        dsEject(data.collection, instance);
+                    // object changed
+                    if (data.action == 'changed') {
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Use `===`.

File: openslides/core/static/js/core/base.js [hunk 3]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
...
+                        changedElements.push(data.data);
                     }
-                    DS.eject(data.collection, data.id);
+                    // object deleted
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Please use `else if` clause for the delete case (also with `===`). If the message is something else the changed or delete we should print a `console.error`.

File: openslides/core/static/js/core/base.js [hunk 4]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
             } catch(err) {
                 console.error(json);
             }
-            _.forEach(dataList, function(data) {
-                console.log("Received object: " + data.collection + ", " + data.id);
-                var instance = DS.get(data.collection, data.id);
-                if (data.action == 'changed') {
+
+            var dataList_byCollection = _.groupBy(dataList, 'collection');
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Proposal: `dataListByCollection`

File: openslides/core/static/js/core/base.js (hunk 3 above)
Comment (emanuelschuetze) [inline openslides/core/static/js/core/base.js]: Done

**Comment by emanuelschuetze** [comment]
@normanjaeckel All done. Ready for review.

File: openslides/core/static/js/core/base.js [hunk 5]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
             } catch(err) {
                 console.error(json);
             }
-            _.forEach(dataList, function(data) {
-                console.log("Received object: " + data.collection + ", " + data.id);
-                var instance = DS.get(data.collection, data.id);
-                if (data.action == 'changed') {
+
+            var dataListByCollection = _.groupBy(dataList, 'collection');
+            _.forEach(dataList_byCollection, function(list, key) {
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Here you also have to use `dataListByCollection`.

File: openslides/core/static/js/core/base.js [hunk 6]
```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
...
-                        // The instance is in the local db
-                        dsEject(data.collection, instance);
+                    // check if object changed or deleted
+                    if (data.action === 'changed') {
+                        changedElements.push(data.data);
+                    } else if (data.action === 'deleted') {
+                        deletedElements.push(data.id);
+                    } else {
+                        console.error('Error: Undefined action for received object' +
+                                '(' + data.collection + ', ' + data.id + ')');
```
Comment (normanjaeckel) [inline openslides/core/static/js/core/base.js]: Indention looks broken.

File: openslides/core/static/js/core/base.js (hunk 5 above)
Comment (emanuelschuetze) [inline openslides/core/static/js/core/base.js]: Done

File: openslides/core/static/js/core/base.js (hunk 6 above)
Comment (emanuelschuetze) [inline openslides/core/static/js/core/base.js]: Done

**Comment by normanjaeckel** [comment]
Rebase and merge.

**Pull request merged**